### ISSUE TITLE
Public DataGridViewCellPaintingEventArgs constructor does not set the `dataGridView` field causing NullRefs

### ISSUE THREAD

**Issue by hughbe** [dotnet/winforms#272]
The following constructor is broken:
```cs
        public DataGridViewCellPaintingEventArgs(DataGridView dataGridView,
                                                 Graphics graphics, 
                                                 Rectangle clipBounds,
                                                 Rectangle cellBounds, 
                                                 int rowIndex, 
                                                 int columnIndex, 
                                                 DataGridViewElementStates cellState,
                                                 object value,
                                                 object formattedValue,
                                                 string errorText,
                                                 DataGridViewCellStyle cellStyle,
                                                 DataGridViewAdvancedBorderStyle advancedBorderStyle,
                                                 DataGridViewPaintParts paintParts)
        {
            if (dataGridView == null)
            {
                throw new ArgumentNullException(nameof(dataGridView));
            }
            if (graphics == null)
            {
                throw new ArgumentNullException(nameof(graphics));
            }
            if (cellStyle == null)
            {
                throw new ArgumentNullException(nameof(cellStyle));
            }
            if ((paintParts & ~DataGridViewPaintParts.All) != 0)
            {
                throw new ArgumentException(string.Format(SR.DataGridView_InvalidDataGridViewPaintPartsCombination, "paintParts"));
            }
            this.graphics = graphics;
            this.clipBounds = clipBounds;
            this.cellBounds = cellBounds;
            this.rowIndex = rowIndex;
            this.columnIndex = columnIndex;
            this.cellState = cellState;
            this.value = value;
            this.formattedValue = formattedValue;
            this.errorText = errorText;
            this.cellStyle = cellStyle;
            this.advancedBorderStyle = advancedBorderStyle;
            this.paintParts = paintParts;
        }
```

The `private DataGridView dataGridView;` field of `DataGridViewCellPaintingEventArgs` is not set so defaults to `null`, so all methods using the args cause a NullReferenceException!!

**Comment by hughbe** [comment]
This contrasts the similar `DataGridViewRowPrePaintEventArgs` and `DataGridViewRowPostPaintEventArgs` which set this field!

**Comment by doMConSwiss** [comment]
Allready fixed by commit e267e523.
https://github.com/dotnet/winforms/commit/e267e523ed516d1c3fc2c331ec90fbd4d9cb3a19
